src/types/any.rs

```diff
@@ -769,16 +746,40 @@ impl PyAny {
         unsafe { ffi::Py_TYPE(self.as_ptr()) }
     }
 
-    /// Casts `self` to a concrete Python object type.
-    ///
-    /// This can cast only to native Python types, not types implemented in Rust.
+    /// Converts this `PyAny` to a concrete Python type.
+    #[deprecated(since = "0.18.0", note = "use the equivalent .downcast()")]
     pub fn cast_as<'a, D>(&'a self) -> Result<&'a D, PyDowncastError<'_>>
     where
         D: PyTryFrom<'a>,
     {
         <D as PyTryFrom<'_>>::try_from(self)
```

Comment:
I think it would be nice if this implementation would forward to `self.downcast` just to make it obvious that they do the same thing.

Reply:
Not possible, since the bounds differ slightly...

Reply:
See https://github.com/PyO3/pyo3/pull/2734#discussion_r1023580363, but then that would be a breaking change even though I don't think the would be real-world breaking due to relaxing the bound.